### PR TITLE
Expand income measure

### PR DESCRIPTION
**Pull request by MattHJensen**
This PR creates a new expanded income measure, `_expanded_income`, and incorporates the new measure as the default for distributional and difference tables. 

This PR is relevant to issue #222.

The intent is that users can still create the tables by AGI like this: `create_distribution_table(calcX, groupby="small_income_bins", result_type="weighted_sum", income_measure='c00100')`. 

On the other hand ` create_distribution_table(calcX, groupby="small_income_bins", result_type="weighted_sum"` would sort the taxpayers by _expanded_income

An important change to note is that any bin with "agi" in the name, I rename to include "income" instead. E.g., `small_agi_bins` is now `small_income_bins`. If merged, this breaks the table creation API and will require updating the webapp. 

The expanded income measure itself includes AGI, employer share of fica, non taxable social security benefits, and non-taxable interest income. There are clearly other items that we might want to add, such as employer-provided health insurance; I included what I thought would be easy to grab from the PUF. @feenberg, could you please review the construction of these variables, below?

    employer_share_fica = (max(0, FICA_ss_trt * min(SS_Earnings_c, e00200) 
                        + FICA_mc_trt * e00200))

    non_taxable_ss_benefits = (e02400 - c02500)

    _expanded_income = (c00100 # AGI
                     + e00400 # Non-taxable interest
                     + non_taxable_ss_benefits
                     + employer_share_fica)

I believe this PR needs review from @SherwinLott, @Amy-Xu, @talumbau, and @feenberg before we merge it.

### CODE REVIEW

**Comment by Amy-Xu** [comment]
The changes look good to me. Do you think posting some tables would be more informational here? For example, tables using c00100 and _expanded_income respectively. It would be very helpful to see the difference.

**Comment by MattHJensen** [comment]
Thanks @Amy-Xu, very good suggestion. I will post distribution tables tabbed by c00100 and by _expanded_income.

**Comment by MattHJensen** [comment]
Here are diagnostic tables (no change, as expected):

Branch:
image

Master:
image

And, @Amy-Xu, here are distribution tables of AGI (modest decrease in progressivity of pre-tax distribution of AGI, as expected): 

Branch
image

Master
image

**Comment by talumbau** [comment]
:+1:, just need to add an issue to the dropQ repo to change to importing the WEBAPP_INCOME_BINS instead of AGI. I'll do that.

**Comment by MattHJensen** [comment]
Just dealt with merge conflicts, so new diagnostics tables: 

PR branch
image

Master
image

**Comment by MattHJensen** [comment]
@talumbau, this is ready to be merged upon your final review. 

Once it is in, we need to do this: https://github.com/OpenSourcePolicyCenter/dropQ/issues/17 and https://github.com/OpenSourcePolicyCenter/webapp/issues/218.

File: taxcalc/functions.py [hunk 1]
```diff
@@ -1427,3 +1427,23 @@ def Taxer_i(inc_in, MARS, II_rt1, II_rt2, II_rt3, II_rt4, II_rt5, II_rt6,
                + II_rt7 * max(0., _a6 - II_brk6[MARS - 1]))
 
     return inc_out
+
+
+@iterate_jit(nopython=True, parameters=['FICA_ss_trt', 'SS_Earnings_c', 
+                                        'FICA_mc_trt'])
```
Comment (talumbau) [inline taxcalc/functions.py]: Parameters are optional now, but you can remove if you desire in a follow-up PR

**Comment by talumbau** [comment]
Looks good to me. I created a "0.5 release" milestone on the webapp and dropq repos for those issues. that's the minor version number next up for taxcalc so it's just a reminder that those need to be done by the time we deploy the next version.

**Comment by MattHJensen** [comment]
Alright. Merging.

**Pull request merged**